bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -93,7 +96,7 @@ private Event createCommandEvent(String topic, String payload, String source) {
         ItemEventPayloadBean bean = deserializePayload(payload, ItemEventPayloadBean.class);
         Command command = null;
         try {
-            command = (Command) parse(bean.getType(), bean.getValue());
+            command = (Command) parse(bean.getType() + TYPE_POSTFIX, bean.getValue());
```

Comment:
Wait, you add the post fix here, because the internal implementation of parse requires it and in order to be able to pass it from the outside you change the parameter from typeName to simpleClassName? Why not keep the signature as is and deal with it only within the parse method?